File: frontend/src/types/ChatMessage.ts
export enum MessageType {
  CHAT = "CHAT",
  IMAGE = "IMAGE",
  JOIN = "JOIN",
  LEAVE = "LEAVE",
  OFFER = "OFFER",
  ANSWER = "ANSWER",
  ICE_CANDIDATE = "ICE_CANDIDATE",
}

export interface ChatMessage {
  id?: string;
  roomId: string;
  senderId: string;
  senderName: string;
  content: string;
  type: MessageType;
  imageUrl?: string;
  timestamp: number;
}

File: backend/src/main/java/com/imap143/domain/entity/ChatMessage.java
package com.imap143.domain.entity;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Data;

@Data
@Document(collection = "messages")
public class ChatMessage {
    @Id
    private String id;
    private String roomId;
    private String senderId;
    private String senderName;
    private String content;
    private String imageUrl;
    private MessageType type;
    private long timestamp;

    public enum MessageType {
        CHAT,           
        JOIN,           
        LEAVE,          
        IMAGE,          // add image type
        OFFER,          // WebRTC offer
        ANSWER,         // WebRTC answer
        ICE_CANDIDATE   // WebRTC ICE candidate
    }
} 